CMakeLists.txt

```diff
@@ -665,6 +682,16 @@ endif()
 # set which precisions to enable
 add_definitions(-DQUDA_RECONSTRUCT=${QUDA_RECONSTRUCT})
 
+# Set target preprocessor flags
+if(${QUDA_TARGET_TYPE} STREQUAL "CUDA")
+  set(QUDA_TARGET_DIR "cuda")
+  add_definitions(-DQUDA_TARGET_QUDA)
+endif()
+if(${QUDA_TARGET_TYPE} STREQUAL "HIP")
+  set(QUDA_TARGET_DIR "hip")
+  add_definitions(-DQUDA_TARGET_HIP)
```

Comment:
I would only want to have that definition on specific source files. Most of QUDA should not care about this and not having it available reduces potential abuse.

Reply:
https://github.com/lattice/quda/commit/087daceb0cd9a69db9a7660d7484d36167869e21